Invoking EstimateGas with nil parameter crashes the node

<!--- First, verify that your issue is not already reported on GitHub -->
<!--- Also test if the latest master branch is affected too -->
<!--- Complete *all* sections as described, this form is processed automatically -->

##### SUMMARY
<!--- Explain the problem briefly below -->
The consensus method EstimateGas crashes when the parameter is nil. This makes the whole node crash. You can invoke this over GRPC.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
<!--- Write the name of the component the issue applies to (e.g. go/worker/storage) below, use your best guess if unsure -->
go consensus. the tendermint/full part, I think

##### OASIS NODE VERSION
<!--- Paste verbatim output from "oasis-node --version" between quotes -->
```paste below
Software version: 20.12.4
Consensus:
  Consensus protocol version: 2.0.0
  Tendermint Core version:    0.34.0
  ABCI library version:       0.17.0
Runtime:
  Host protocol version:      1.0.0
  Committee protocol version: 1.0.0
Go toolchain version: 1.15.5
```

##### OS / ENVIRONMENT
<!--- Provide all relevant information below, e.g. which distribution and version, which Go version, which Rust version -->
gnu/linux--should not be related

##### STEPS TO REPRODUCE
<!--- Describe exactly how to reproduce the problem, using a minimal test-case -->
you'll need to make a grpc client. call EstimateGas with a nil request

##### ACTUAL RESULTS
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->

<!--- Paste verbatim command output between quotes -->
```paste below
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x10a8350]

goroutine 1212337 [running]:
github.com/oasisprotocol/oasis-core/go/consensus/tendermint/abci.(*abciMux).EstimateGas(0xc0001d6c40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        github.com/oasisprotocol/oasis-core/go/consensus/tendermint/abci/mux.go:597 +0xd0
github.com/oasisprotocol/oasis-core/go/consensus/tendermint/abci.(*ApplicationServer).EstimateGas(...)
        github.com/oasisprotocol/oasis-core/go/consensus/tendermint/abci/mux.go:177
github.com/oasisprotocol/oasis-core/go/consensus/tendermint/full.(*fullService).EstimateGas(0xc0002cf6c0, 0x1f539e0, 0xc01a4ef890, 0xc01a4ef7a0, 0xc00
02cf6c0, 0x1969a00, 0x1bb1b20)
        github.com/oasisprotocol/oasis-core/go/consensus/tendermint/full/full.go:520 +0x6c
github.com/oasisprotocol/oasis-core/go/consensus/api.handlerEstimateGas.func1(0x1f539e0, 0xc01a4ef890, 0x1847600, 0xc01a4ef7a0, 0x1847600, 0xc01a4ef7a
0, 0x0, 0x0)
        github.com/oasisprotocol/oasis-core/go/consensus/api/grpc.go:229 +0x8d
...
```

##### EXPECTED RESULTS
<!--- Describe what you expected to happen when running the steps above -->
probably just indicate an error in the response and continue running

--- a/.changelog/3648.bugfix.md
+++ b/.changelog/3648.bugfix.md
@@ -1,0 +1,1 @@
+go/consensus/tendermint: Fix nil deref in EstimateGas

--- a/go/consensus/api/api.go
+++ b/go/consensus/api/api.go
@@ -54,6 +54,9 @@ var (
 
 	// ErrDuplicateTx is the error returned when the transaction already exists in the mempool.
 	ErrDuplicateTx = errors.New(moduleName, 5, "consensus: duplicate transaction")
+
+	// ErrInvalidArgument is the error returned when the request contains an invalid argument.
+	ErrInvalidArgument = errors.New(moduleName, 6, "consensus: invalid argument")
 )
 
 // FeatureMask is the consensus backend feature bitmask.

--- a/go/consensus/tendermint/abci/mux.go
+++ b/go/consensus/tendermint/abci/mux.go
@@ -573,6 +573,10 @@ func (mux *abciMux) executeTx(ctx *api.Context, rawTx []byte) error {
 }
 
 func (mux *abciMux) EstimateGas(caller signature.PublicKey, tx *transaction.Transaction) (transaction.Gas, error) {
+	if tx == nil {
+		return 0, consensus.ErrInvalidArgument
+	}
+
 	// As opposed to other transaction dispatch entry points (CheckTx/DeliverTx), this method can
 	// be called in parallel to the consensus layer and to other invocations.
 	//

--- a/go/consensus/tests/tester.go
+++ b/go/consensus/tests/tester.go
@@ -94,6 +94,9 @@ func ConsensusImplementationTests(t *testing.T, backend consensus.ClientBackend)
 	require.NoError(err, "GetEpoch")
 	require.True(epoch > 0, "epoch height should be greater than zero")
 
+	_, err = backend.EstimateGas(ctx, &consensus.EstimateGasRequest{})
+	require.ErrorIs(err, consensus.ErrInvalidArgument, "EstimateGas with nil transaction should fail")
+
 	_, err = backend.EstimateGas(ctx, &consensus.EstimateGasRequest{
 		Signer:      memorySigner.NewTestSigner("estimate gas signer").Public(),
 		Transaction: transaction.NewTransaction(0, nil, staking.MethodTransfer, &staking.Transfer{}),

